src/inference/Inference.jl

```diff
@@ -130,6 +130,12 @@ struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
     stat  :: S
 end
 
+function Transition(vi::AbstractVarInfo, t)
+    theta = tonamedtuple(vi)
+    lp = getlogp(vi)
+    return Transition(theta, lp, getstats(t))
```

Comment:
Could we define a default fallback
```
getstats(t) = nothing
```
here? Then one could read and understand this method, and `getstats` would be extended in `src/contrib/inference/abstractmcmc.jl`. Additionally, we could set `t = nothing` as default argument above and unify it with the `Transition` constructor below.

BTW I think it is wrong that #2008 used a default `getstats(t) = NamedTuple()` since this is inconsistent with the existing `Transition` constructor below.

Reply:
Nice suggestions! 
Yes, I am also a little puzzled now that you point it out: @torfjelde do we have a good reason to default `getstats(t) = NamedTuple()` in contrib/abstractmcmc.jl?

Reply:
Exactly. My intention was to implement the external sampler stuff without touching any other code, deferring these sort of changes to another PR. But given that we didn't make a release, we probably could have just put them all in the same PR :grimacing: